spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/BatchMessagingMessageListenerAdapter.java

```diff
@@ -117,9 +117,10 @@ public void setBatchToRecordAdapter(BatchToRecordAdapter<K, V> batchToRecordAdap
 	 * properly propagated to the batch converter's record converter for message conversion
 	 * in batch listeners.
 	 * <p>
-	 * Uses the same validation as the parent class to prevent the paradox of choice:
-	 * not allowed when a custom {@link #setBatchMessageConverter(BatchMessageConverter)
-	 * batchMessageConverter} is provided. Since {@link BatchMessagingMessageConverter} now
+	 * This method cannot be called after {@link #setBatchMessageConverter(BatchMessageConverter)
+	 * setBatchMessageConverter()} as it would cause a mutation of the internal
+	 * batchMessageConverter. Instead, the SmartMessageConverter has to be provided on the
+	 * external BatchMessageConverter. Since {@link BatchMessagingMessageConverter} now
```

Comment:
Please, revise this Javadoc for code snippets.
I believe that `setBatchMessageConverter()` link is a bit off.
The `batchMessageConverter` as to be link, as well as `SmartMessageConverter` and `BatchMessageConverter`.